### ISSUE TITLE
Can't colorize a part of of a Jenkins log.

### ISSUE THREAD

**Issue by daxgames** [jenkinsci/ansicolor-plugin#180]
Please provide precise details of the issue you detected.

### Test environment

- Jenkins Version: 2.222.3
- AnsiColor Version: 0.6.3
- Hostsystem: Linux

### Expected behavior
What did you expect the plugin to do?

Colorize the ansiColor wrapped shell script output ONLY.

### Actual behavior
What did the plugin do instead?

The entire log is colorized.

### Steps to reproduce the behavior

Create a pipeline job with the below content:

```
pipeline {
    agent { node { label "master" } }
    stages {
        stage('job') {
            steps {
                script {
                    echo "Before Ansi Color Wrap."
                    ansiColor('vga') {
                        sh "ls -la"
                    }
                    echo "After Ansi Color Wrap."
                }
            }
        }
    }
}
```

**Comment by daxgames** [comment]
This used to work as suggested.

**Comment by tszmytka** [comment]
I can reproduce with the provided snippet. 
Additionally it is not possible to use multiple `ansiColor` steps having different color maps - a use case maybe exotic but fully doable in a pipeline script. Currently the plugin utilizes the last-one-wins strategy.

I was thinking about equipping the `ColorizedAction` built in 
https://github.com/jenkinsci/ansicolor-plugin/blob/b1336fdd75eb66bcc21cf7a6384f3ef6c028738d/src/main/java/hudson/plugins/ansicolor/AnsiColorStep.java#L74-L83
with the line numbers which are supposed to be processed and then using that information (later - when output is rendered) in `hudson.plugins.ansicolor.ColorConsoleAnnotator#annotate`. Not sure still if getting that information would be possible.
I would appreciate a second opinion on this. Am I not re-inventing the wheel here? Is there a mechanism in Jenkins that does something similar?

**Comment by daxgames** [comment]
The exact issue also happens in a scripted pipeline.  I supplied a declarative because that is what I had.  I have never tried doing multiple colormaps in either type of pipeline.  I prefer the vga colormap because if it were working properly it really highlights the specific area of the log that is colorized because of the black background.

**Comment by dwnusbaum** [comment]
> I was thinking about equipping the ColorizedAction built in with the line numbers which are supposed to be processed and then using that information (later - when output is rendered) in hudson.plugins.ansicolor.ColorConsoleAnnotator#annotate. Not sure still if getting that information would be possible.

I do not think that tracking line numbers this way would be possible, and even if it is, I would not try to take that approach. 

#132 (0.6.0+) is the reason why this does not work anymore (this behavior is what [this code comment](https://github.com/jenkinsci/ansicolor-plugin/blob/b1336fdd75eb66bcc21cf7a6384f3ef6c028738d/src/main/java/hudson/plugins/ansicolor/ColorizedAction.java#L31) refers to, see also [this comment](https://github.com/jenkinsci/ansicolor-plugin/pull/132#discussion_r230945123) in the PR), and I am not sure if there is a way to fix it while still having coloring work for logs produced by steps like `sh` and `bat`, which is the reason for that PR. If you do want to fix it, you would need to use an API that can be passed to `context.newBodyInvoker().withContext` (e.g. `ConsoleLogFilter`) so that the coloring only applies to log content produced inside of that step, but you would also need an API that runs on the master (`ConsoleLogFilter` runs remotely on the agent for `sh` and `bat` steps). `ConsoleAnnotator` is currently used because it runs on the master, but it is not contextual, it applies to the whole build. 

I don't think there is currently an API that can be used to change this behavior without introducing other issues.

**Comment by daxgames** [comment]
Is this supposed to be fixed in 0.7.x Still seeing it in 0.7.3.
